gateway/crates/gateway-binary/tests/telemetry/metrics/operation/hooks.rs

```diff
@@ -0,0 +1,162 @@
+use crate::telemetry::metrics::{with_custom_gateway, ExponentialHistogramRow, METRICS_DELAY};
+
+#[test]
+fn on_gateway_request_success() {
+    let config = indoc::indoc! {r#"
+        [hooks]
+        location = "../../../engine/crates/wasi-component-loader/examples/target/wasm32-wasip1/debug/gateway_request_no_op.wasm"
```

Comment:
When are these built?

Reply:
https://github.com/grafbase/grafbase/blob/main/.github/actions/test/action.yml#L68-L74

Reply:
Benjamin did a similar thing in the integration tests.